### PR TITLE
clio-38286: Upgrade Angular to 1.5.7

### PR DESCRIPTION
**Pull request by JefCurtis**
- Update Angular from 1.4.7 to 1.5.7
- Update ui-select from 0.13.1 to 0.18.0
- Fixed for ui breaking changes in ui-select
- Fix for several directives where isolated scoped variable should be
optional

### CODE REVIEW

File: themis_components/thAutocomplete/index.scss [hunk 1]
```diff
@@ -179,11 +179,25 @@ th-autocomplete {
...
-    }
-
-    input {
       font-size: inherit;
       background: $off-white;
+
+      min-width: 100%;
+
+      padding: 12px;
+      padding-right: 40px;
```
Comment (dtomalty) [inline themis_components/thAutocomplete/index.scss]: Would prefer these on one line, so one isn't missed if they need to be changed.

Comment (dtomalty) [inline themis_components/thAutocomplete/index.scss]: Also, right padding used to take into account the icon. Does it still?

File: themis_components/thAutocomplete/index.scss [hunk 2]
```diff
@@ -179,11 +179,25 @@ th-autocomplete {
 
     .ui-select-search {
       font-family: inherit;
-    }
-
-    input {
       font-size: inherit;
       background: $off-white;
+
+      min-width: 100%;
```
Comment (dtomalty) [inline themis_components/thAutocomplete/index.scss]: What is this used for?

File: themis_components/thFilter/filters/number/thFilter.number.directive.spec.coffee [hunk 3]
```diff
@@ -85,7 +85,7 @@ describe "ThemisComponents: Directive: thFilterNumber", ->
         setInputValue "1000"
 
       it "should call onFilterChange", ->
-        selectOperator operatorOptions[1].value
+        selectOperator operatorOptions[1]
```
Comment (dtomalty) [inline themis_components/thFilter/filters/number/thFilter.number.directive.spec.coffee]: There's another instance in the previous test..

File: themis_components/thAutocomplete/index.scss (hunk 1 above)
Comment (JefCurtis) [inline themis_components/thAutocomplete/index.scss]: It does accommodate an icon but left the padding in regardless if there was an icon or not. Updated to add padding conditionally.

File: themis_components/thAutocomplete/index.scss (hunk 2 above)
Comment (JefCurtis) [inline themis_components/thAutocomplete/index.scss]: This was causing crazy resizing with the window observer. By setting it to have a min width of 100% there was no need for it to resize.

File: themis_components/thAutocomplete/index.scss (hunk 1 above)
Comment (dtomalty) [inline themis_components/thAutocomplete/index.scss]: :white_check_mark: Resolved

File: themis_components/thAutocomplete/index.scss (hunk 2 above)
Comment (dtomalty) [inline themis_components/thAutocomplete/index.scss]: :white_check_mark: Resolved

File: themis_components/thFilter/filters/number/thFilter.number.directive.spec.coffee (hunk 3 above)
Comment (dtomalty) [inline themis_components/thFilter/filters/number/thFilter.number.directive.spec.coffee]: :white_check_mark: Resolved

**Pull request merged**